src/App.jsx

```diff
@@ -14,70 +16,88 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  function setHeading(selectedGood) {
+    return goods.includes(selectedGood)
+      ? `${selectedGood} is selected`
+      : 'No goods selected';
+  }
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const [selectedGood, setselectedGood] = useState('Jam');
+  let key = 0;
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood ? (
+          setHeading(selectedGood)
+        ) : (
+          setHeading(selectedGood)
+        )}
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        {goods.includes(selectedGood) && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
+              setselectedGood('');
+            }}
+          />
+        )}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const isSelected = selectedGood === good;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+            key += 1;
```

Comment:
Bad practice🚨 Your key is recalculated on each render which makes it invalid. Better use `uuid` library or pregenerate ids once (outside of App component). In this task particularly you can just use good name as a key